apps/site/src/_pages/docs/4_spec/04_service-module.mdx

```diff
@@ -0,0 +1,50 @@
+# Service Module
+
+This document (version 0.1) defines a **service module** according to the terms set out in the [core specification](https://blockprotocol.org/docs/spec/core).
+
+The service module enables blocks to interact with external services without needing to be provided with credentials: the embedding application calls the external service on the block's behalf.
+
+## Glossary
+
+**Provider:** a company, organization or person that provides a service or services (e.g. Mapbox)
+
+**Service:** a request handler which meets a specific functional goal, typically external to the embedding application and accessed via third-party API endpoint or integration (e.g. Retrieve Address).
+
+## Messages
+
+### Handling messages
+
+### <Hidden>Handling messages </Hidden>Summary
+
+The service module specifies named messages from named service providers, and any messages exchanged between application and block should be consistent with the shape and intent of those messages.
+
+How exactly the embedding application generates those messages is beyond the scope of the Block Protocol, so long as the schema and intent are met.
+
+#### <Hidden>Handling messages </Hidden>Specification
+
+Embedding applications MUST ensure that any responses they provide to blocks to messages sent under the service module are consistent with the schema and described intent of the message in this specification, and consistent with the response that the named service provider would return, except for any fields which are required by the named service provider but omitted from this specification (e.g. because they relate to authentication or other information that the block should not or does not have access to, and is added by the embedding application when making an onward request).
+
+They MAY implement messages by any of:
+
+- calling the named service provider directly, OR
+- generating the response internally, OR
+- calling another third-party that provides an appropriate response, whether directly or by themselves making a request to the named service provider.
+
+### Error definitions
+
+#### <Hidden>Error definitions </Hidden>Specification
+
+If a request fails, embedding applications MUST provide blocks with an error which complies with the specification, i.e. contains a listed code and a message.
+
+Embedding applications MAY withhold or translate error codes and messages sent by the service provider when reporting an error to the block.
```

Comment:
```suggestion
Embedding applications MAY withhold or translate error codes and messages sent by the service provider when reporting an error to the block, while still ensuring the error complies with the specification.
```

Reply:
Can we also be clear about what "the specification" is. Is it "the definition of the message in the service module specification"? Is it the external provider specification? Is it something else?

Reply:
https://github.com/blockprotocol/blockprotocol/pull/1049/commits/d41705f36b21d5a368ddff99f5c7388bc77ccd95